### PR TITLE
Poetry setup

### LINKED ISSUE #5: Setup poetry
Currently, we use the redundant method to install packages via requirements.txt
Improve this by setting up the entire project with poetry.

### PR DESCRIPTION
**Pull request by jsparmani**
This fixes #5 
@shubhank-saxena I have setup poetry to manage dependencies, also `pyproject.toml` itself is handling black configs, somehow flake8 still doesn't support `.toml` config. Please have a look and let me know if I should start updating `CONTRIBUTION.md` as well.

### CODE REVIEW

**Comment by shubhank-saxena** [comment]
@jsparmani why do you need flake8 in poetry file when it is a pre-commit? Any special configurations that you desire for?

**Comment by jsparmani** [comment]
@shubhank-saxena I read that poetry is used to pack dependencies in a better way and reduce the number of files required for configuration into a single file to make it developer friendly that's why.
I used this article as reference (https://dev.to/yukinagae/beginner-guide-on-poetry-new-python-dependency-management-tool-4327)

**Comment by jsparmani** [comment]
@shubhank-saxena Actually I was checking out this article (https://dev.to/yukinagae/beginner-guide-on-poetry-new-python-dependency-management-tool-4327), it talks about how poetry brings all configs inside a single file and packs all the project dependencies, this article also suggested removing `setup.cfg` . So, I tried that way.

File: pyproject.toml [hunk 1]
```diff
@@ -1,3 +1,55 @@
...
+toml = "0.10.0"
+urllib3 = "1.25.9"
+virtualenv = "20.0.17"
+wincertstore = "0.2"
+
+
+[tool.poetry.dev-dependencies]
+
+[build-system]
+requires = ["poetry>=0.12"]
```
Comment (shubhank-saxena) [inline pyproject.toml:50]: You may remove this line! Let flake8 be automatically handled by pre-commit.

File: pyproject.toml [hunk 2]
```diff
@@ -1,3 +1,55 @@
...
+build-backend = "poetry.masonry.api"
+
 [tool.black]
```
Comment (shubhank-saxena) [inline pyproject.toml:53]: Also, make sure you first delete the existing pyproject.toml file

Comment (shubhank-saxena) [inline pyproject.toml:53]: add one more [tool.flake8] and use the following
```python
[tool.flake8]
exclude = */.*/,.*/,__pycache__,node_modules
# E129: visually indented line with same indent as next logical line
# E203 whitespace before ':'
# E231: missing whitespace after ','
# E501: line too long
extend_ignore = E129,E203,E231,E501
max-line-length = 100
```

File: pyproject.toml (hunk 1 above)
Comment (jsparmani) [reply pyproject.toml:50]: Okay!

File: pyproject.toml (hunk 2 above)
Comment (jsparmani) [reply pyproject.toml:53]: Yes

Comment (jsparmani) [reply pyproject.toml:53]: Do, we require a `setup.cfg` file after these configs?

**Comment by shubhank-saxena** [comment]
> @shubhank-saxena Actually I was checking out this article (https://dev.to/yukinagae/beginner-guide-on-poetry-new-python-dependency-management-tool-4327), it talks about how poetry brings all configs inside a single file and packs all the project dependencies, this article also suggested removing `setup.cfg` . So, I tried that way but then flake8 docs say that `.toml` files aren't supported.

It does support I guess. 
https://dev.to/raiderrobert/python-tooling-review-pyproject-toml-and-flake8-45bb

File: pyproject.toml (hunk 2 above)
Comment (shubhank-saxena) [reply pyproject.toml:53]: Nopes. Delete that too

**Comment by shubhank-saxena** [comment]
Is it working?

**Comment by jsparmani** [comment]
> Is it working?

Yeah, it's working this way! Thanks!

**Comment by shubhank-saxena** [comment]
How did you configure Django to pickup poetry installation?

**Comment by jsparmani** [comment]
> How did you configure Django to pickup poetry installation?

Django is installed in the `env/` folder made by `virtualenv`. Dependencies in `env/ ` have been installed using `poetry install`

**Comment by jsparmani** [comment]
@shubhank-saxena Is there any specific way to connect Django and poetry. Currently, poetry just provides functionality to maintain project dependencies similar to `package.json` in js.

**Pull request merged**